src/System.Private.CoreLib/src/System/String.Searching.cs

```diff
@@ -88,17 +88,100 @@ public int IndexOfAny(char[] anyOf, int startIndex)
             return IndexOfAny(anyOf, startIndex, this.Length - startIndex);
         }
 
-        public unsafe int IndexOfAny(char[] anyOf, int startIndex, int count)
+        public int IndexOfAny(char[] anyOf, int startIndex, int count)
         {
             if (anyOf == null)
                 throw new ArgumentNullException(nameof(anyOf));
 
-            if (startIndex < 0 || startIndex > Length)
+            if ((uint)startIndex > (uint)Length)
                 throw new ArgumentOutOfRangeException(nameof(startIndex), SR.ArgumentOutOfRange_Index);
 
-            if (count < 0 || count > Length - startIndex)
+            if ((uint)count > (uint)(Length - startIndex))
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_Count);
 
+            if (anyOf.Length == 2)
+            {
+                // Very common optimization for directory separators (/, \), quotes (", '), brackets, etc
+                return IndexOfAny(anyOf[0], anyOf[1], startIndex, count);
+            }
+            else if (anyOf.Length == 3)
+            {
+                return IndexOfAny(anyOf[0], anyOf[1], anyOf[2], startIndex, count);
+            }
+            else if (anyOf.Length > 3)
+            {
+                return IndexOfCharArray(anyOf, startIndex, count);
+            }
+            else if (anyOf.Length == 1)
+            {
+                return IndexOf(anyOf[0], startIndex, count);
+            }
+            else // anyOf.Length == 0
+            {
+                return -1;
+            }
+        }
+
+        private unsafe int IndexOfAny(char value1, char value2, int startIndex, int count)
+        {
+            fixed (char* pChars = &_firstChar)
+            {
+                char* pCh = pChars + startIndex;
+
+                while (count > 0)
+                {
+                    char c = *pCh;
+
+                    if (c == value1 || c == value2)
+                        goto ReturnIndex;
```

Comment:
Could you please incorporate https://github.com/dotnet/coreclr/pull/13505 to this change as well?

Reply:
Oh missed that one getting merged. Done

Reply:
Note String.Comparison.cs is different in RT and that code port was not needed.